### PR TITLE
[Merged by Bors] - Conversion of `ResMut` and `NonSendMut` to `Mut`

### PR DESCRIPTION
**Pull request by djeedai**
# Objective

Enable treating components and resources equally, which can
simplify the implementation of some systems where only the change
detection feature is relevant and not the kind of object (resource or
component).

## Solution

Implement `From<ResMut<T>>` and `From<NonSendMut<T>>` for
`Mut`. Since the 3 structs are similar, and only differ by their system
param role, the conversion is trivial.

---

## Changelog

Added - `From<ResMut>` and `From<NonSendMut>` for `Mut<T>`.

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
I like the idea! IMO this should just be a `From` trait implementation though; that captures the semantics exactly and is more convenient and conventional.

**Review by alice-i-cecile — APPROVED** [review]

**Review by maniwani — APPROVED** [review]

File: crates/bevy_ecs/src/change_detection.rs [hunk 1]
```diff
@@ -206,6 +217,17 @@ change_detection_impl!(NonSendMut<'a, T>, T,);
 impl_into_inner!(NonSendMut<'a, T>, T,);
 impl_debug!(NonSendMut<'a, T>,);
 
+impl<'a, T: Resource> From<NonSendMut<'a, T>> for Mut<'a, T> {
```
Comment (djeedai) [inline crates/bevy_ecs/src/change_detection.rs:220]: This is wrong, should have been:
```suggestion
impl<'a, T: 'static> From<NonSendMut<'a, T>> for Mut<'a, T> {
```
FYI @alice-i-cecile , @maniwani

Comment (djeedai) [reply crates/bevy_ecs/src/change_detection.rs:220]: Logged as #5456

**Pull request closed without merging**